wb_to_prepbufr.py

```diff
@@ -207,71 +321,115 @@ def convert_to_prepbufr(data, output_file='export.prepbufr'):
-        observations_page = wb_get_request(
-            f"https://sensor-data.windbornesystems.com/api/v1/super_observations.json?since={since}&include_mission_name=true")
-
+        observations_page = wb_get_request(next_page)
+        has_next_page = observations_page["has_next_page"]
+        if (len(observations_page['observations']) == 0):
+            print("Could not find any observations for the input date range!!!!")
+        if has_next_page:
+            next_page = observations_page["next_page"]+"&include_mission_name=true&max_time={}".format(endtime)
         print(f"Fetched page with {len(observations_page['observations'])} observation(s)")
         for observation in observations_page['observations']:
-            if observation['mission_name'] not in observations_by_mission:
+            if 'mission_name' not in observation:
+                print("got an ob without a mission name???")
+                continue
+            elif observation['mission_name'] not in observations_by_mission:
                 observations_by_mission[observation['mission_name']] = []
 
             observations_by_mission[observation['mission_name']].append(observation)
+            accumulated_observations.append(observation)
 
-        # if there's no new data, break
-        if not observations_page['has_next_page']:
-            break
-
-            # alternatively, you could call `time.sleep(60)` and keep polling
+            # alternatively, you could call `time.sleep(60)` and keep polling here
             # (though you'd have to move where you were calling convert_to_prepbufr)
 
-        # update since for the next request
-        since = observations_page['next_since']
 
     if len(observations_by_mission) == 0:
         print("No observations found")
         return
 
-    for mission_name, accumulated_observations in observations_by_mission.items():
-        # make sure it's sorted
-        accumulated_observations.sort(key=lambda x: x['timestamp'])
-
-        # slice into 3 hour segments
-        start_index = 0
-        for i in range(len(accumulated_observations)):
-            if accumulated_observations[i]['timestamp'] - accumulated_observations[start_index]['timestamp'] > BUCKET_HOURS * 60 * 60:
-                segment = accumulated_observations[start_index:i]
-                output_file = f"windborne_data_{segment[0]['timestamp']}.prepbufr"
-
-                print(f"Converting {len(segment)} observation(s) to prepbufr and saving as {output_file}")
-                convert_to_prepbufr(segment, output_file)
-
-                start_index = i
-
-        segment = accumulated_observations[start_index:]
-        output_file = f"windborne_data_{mission_name}_{segment[0]['timestamp']}.prepbufr"
-        print(f"Converting {len(segment)} observation(s) to prepbufr and saving as {output_file}")
-        convert_to_prepbufr(segment, output_file)
-
+    if (not args.split_by_mission):
```

Comment:
If we keep splitting by mission defaulting to false: I'd recommend that we flip the order here to avoid an if (not) / else, just for the sake of clarity.